File: javascript/Day_3/Day_3.js
/**
 * Problème
 * -------
 * Écrire un programme qui calcule le prix de vente TTC (toutes taxes comprises) d'un article
 * dont on connaît le prix unitaire et la quantité.
 * Si le prix de vente TTC est supérieur à 100, on accorde une réduction de 10%.
 * Le programme est protégé par un mot de passe 'anne' et la TVA est de 16%.
 *
 * Catégorie
 * ---------
 * Numbers
 *
 * Niveau
 * ------
 * Débutant
 *
 * @author
 * 10Glo (https://github.com/10Glo)
 * @version 1.0
 * @since 17/09/2025
 */

const readline = require("readline");

// --- Setup de readline ---
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
});

// --- Variables ---
let unitPrice = 0,
    salesPriceIncludingAllTaxes = 0,
    salesPriceExcludingTax = 0;
let quantity = 0;
let productName = "";
const TVA = 0.16;

// --- Titre du projet ---
console.log("==============================================");
console.log(
    ":::::::::      :::   :::   :::      ::::::::  \n" +
    ":+:    :+:   :+: :+: :+:   :+:     :+:    :+: \n" +
    "+:+    +:+  +:+   +:+ +:+ +:+             +:+ \n" +
    "+#+    +:+ +#++:++#++: +#++:           +#++:  \n" +
    "+#+    +#+ +#+     +#+  +#+               +#+ \n" +
    "#+#    #+# #+#     #+#  #+#        #+#    #+# \n" +
    "#########  ###     ###  ###         ########  "
);
console.log("==============================================\n");

// --- Mot de passe ---
rl.question("🔑 Veuillez entrer le mot de passe : ", (password) => {
    if (password !== "anne") {
        console.log("❌ Le mot de passe n'est pas correct !");
        console.log("\nVeuillez recommencer !!!");
        rl.close();
        console.log("\nFin du programme ✅");
        process.exit(0);
    }

    console.log("✅ Le mot de passe est correct\n");

    // --- Nom du produit ---
    rl.question("➡️  Entrez le nom de l'article : ", (name) => {
        productName = name;

        // --- Prix unitaire ---
        rl.question(
            `➡️  Entrez le prix unitaire de l'article ${productName} --> `,
            (unit) => {
                unitPrice = parseFloat(unit);
                if (isNaN(unitPrice)) {
                    console.log(
                        "\n⚠️ Erreur : Vous devez entrer un nombre valide pour le prix unitaire !"
                    );
                    rl.close();
                    console.log("\nFin du programme ✅");
                    process.exit(0);
                }

                // --- Quantité ---
                rl.question(
                    `➡️  Entrez la quantité de l'article ${productName} --> `,
                    (qty) => {
                        quantity = parseInt(qty);
                        if (isNaN(quantity)) {
                            console.log(
                                "\n⚠️ Erreur : Vous devez entrer un nombre valide pour la quantité !"
                            );
                            rl.close();
                            console.log("\nFin du programme ✅");
                            process.exit(0);
                        }

                        // --- Résumé ---
                        console.log("\n-------------------------------------");
                        console.log(" 📋 Résumé des informations saisies");
                        console.log("-------------------------------------");
                        console.log(`🔠    Le nom de l'article : ${productName}`);
                        console.log(`🔢    Le prix unitaire : ${unitPrice}`);
                        console.log(`🔢    La quantité : ${quantity}\n`);

                        // --- Calculs ---
                        console.log("--------------");
                        console.log(" 📋 Résultats");
                        console.log("--------------\n");

                        salesPriceExcludingTax = unitPrice * quantity;
                        console.log(
                            `➡️  Le prix de vente hors taxes est  --> ${salesPriceExcludingTax.toFixed(
                                2
                            )} CDF\n`
                        );

                        console.log("➡️  La TVA (Taxe sur la Valeur Ajoutée) est de 16% ");
                        salesPriceIncludingAllTaxes = salesPriceExcludingTax * (1 + TVA);
                        console.log(
                            `➡️  Le prix de vente toutes taxes comprises est  --> ${salesPriceIncludingAllTaxes.toFixed(
                                2
                            )} CDF\n`
                        );

                        // --- Réduction ---
                        if (salesPriceIncludingAllTaxes > 100) {
                            const reduction = salesPriceIncludingAllTaxes * 0.1;
                            const finalPrice = salesPriceIncludingAllTaxes * 0.9;
                            console.log(
                                `➡️  Une réduction de 10% (${reduction.toFixed(
                                    2
                                )} CDF) est appliquée au prix TTC (${salesPriceIncludingAllTaxes.toFixed(
                                    2
                                )})\nVoici le prix de vente final  --> ${finalPrice.toFixed(
                                    2
                                )} CDF\n`
                            );
                        }

                        // --- Fin ---
                        rl.close();
                        console.log("Fin du programme ✅");
                        console.log("-------------------------------------------");
                    }
                );
            }
        );
    });
});

File: java/Day_3/src/Day_3.java
import java.text.DecimalFormat;
import java.util.Scanner;

/**
 * Problème
 * -------
 * Ecrire un programme qui calcule le prix de vente TTC (toutes taxes comprises) d'un article dont on connait le prix unitaire et la quantité.
 * Si le prix de vente TTC (toutes taxes comprises) est superieur  à 100, on accorde une réduction de 10%.
 * Le programme est protégé par un mot de passe 'anne' et Le TVA est de 16%.
 *
 * Catégorie
 * ---------
 * Numbers
 * 
 * Niveau
 * ------
 * Débutant
 *
 * @author 10Glo (https://github.com/10Glo)
 * @version 1.0
 * @since 17/09/2025
 */

public class Day_3 {

    public static void main(String[] args) {

        // --- Déclaration des Variables ---
        Scanner scanner = new Scanner(System.in);
        double unitPrice = 0, salesPriceIncludingAllTaxes = 0, salesPriceExcludingTax = 0;
        int quantity = 0;

        // formatteur pour 2 décimales
        DecimalFormat df = new DecimalFormat("#.##");

        String password = "", productName = "";

        // --- Titre du projet ---
        System.out.println("==============================================");
        System.out.println(":::::::::      :::   :::   :::      ::::::::  \n" +
                ":+:    :+:   :+: :+: :+:   :+:     :+:    :+: \n" +
                "+:+    +:+  +:+   +:+ +:+ +:+             +:+ \n" +
                "+#+    +:+ +#++:++#++: +#++:           +#++:  \n" +
                "+#+    +#+ +#+     +#+  +#+               +#+ \n" +
                "#+#    #+# #+#     #+#  #+#        #+#    #+# \n" +
                "#########  ###     ###  ###         ########  ");
        System.out.println("==============================================\n");

        // --- Saisie du Mot de Passe ---
        System.out.println("\uD83D\uDD11️  Veuillez entrer le mot de passe :  ");
        password = scanner.nextLine();
        if (password.equals("anne"))
            System.out.println("Le mot de passe est correct ✅\n");
        else {
            System.out.println("Le mot de passe n'est pas correct ❌");
            System.out.println("\nVeuillez Recommencer !!!");

            scanner.close();
            System.out.println("\nFin du programme ✅");
            System.exit(0);
        }


        // --- Saisie des informations ---
        try {
            System.out.print("➡️  Entrez le nom de l'article : ");
            productName = scanner.nextLine();

            System.out.print("➡️  Entrez le prix unitaire de l'article " + productName + " --> ");
            unitPrice = scanner.nextInt();

            System.out.print("➡️  Entrez la quantité de l'article " + productName + " --> ");
            quantity = scanner.nextInt();
        } catch (Exception exception) {
            System.out.println("\n⚠️ Erreur : Vous devez entrer un nombre valide pour le prix unitaire et la quantité de l'article !\n");

            scanner.close();
            System.out.println("\nFin du programme ✅");
            System.exit(0);
        }

        // --- Résumé des informations ---
        System.out.println("\n-------------------------------------");
        System.out.println(" 📋 Résumé des informations saisies");
        System.out.println("-------------------------------------");
        System.out.println("\uD83D\uDD20    Le nom de l'article : " + productName + "\n");
        System.out.println("\uD83D\uDD22    Le prix unitaire : " + unitPrice + "\n");
        System.out.println("\uD83D\uDD22    La quantité : " + quantity + "\n");

        // --- Résutats ---
        System.out.println("--------------");
        System.out.println(" 📋 Résutats");
        System.out.println("--------------\n");

        salesPriceExcludingTax = unitPrice * quantity;
        System.out.println("➡️  Le prix de vente hors taxes est  --> " + df.format(salesPriceExcludingTax) + " CDF\n");
        System.out.println("➡️  Le TVA(Taxe sur la Valeur Ajoutée) est de 16% ");
        salesPriceIncludingAllTaxes = salesPriceExcludingTax * (1 + 0.16);
        System.out.println("➡️  Le prix de vente toutes taxes comprises est  --> " + df.format(salesPriceIncludingAllTaxes) + " CDF\n");

        // --- Réduction ---
        if (salesPriceIncludingAllTaxes > 100)
            System.out.println("➡️  Une Réduction de 10% (" + df.format(salesPriceIncludingAllTaxes * 0.10) + " cdf) est appliqué au prix de vente toutes taxes comprises (" + df.format(salesPriceIncludingAllTaxes) + ") \nVoici le prix de vente final  --> " + df.format(salesPriceIncludingAllTaxes *  0.90) + " CDF\n");

        // --- Message de fin ---
        scanner.close();
        System.out.println("Fin du programme ✅");
        System.out.println("-------------------------------------------");

    }
}
